File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryDisplayCodeTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class DeliveryDisplayCodeTest {
  @Test
  fun `formats store daily number with leading zeros`() {
    assertEquals("N. dia 009", deliveryDailyNumberLabel(9))
  }

  @Test
  fun `uses daily number as primary label when available`() {
    assertEquals("N. dia 009", deliveryPrimaryCodeLabel("AST-20260517-009", 9))
  }

  @Test
  fun `falls back to public code for older deliveries`() {
    assertEquals(null, deliveryDailyNumberLabel(null))
    assertEquals("ENT-000001", deliveryPrimaryCodeLabel("ENT-000001", null))
  }
}

File: apps/motoboy-pwa/src/deliveryDisplayCode.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { deliveryDailyNumberLabel, deliveryPrimaryCodeLabel } from "./deliveryDisplayCode";

test("formats store daily number with leading zeros", () => {
  assert.equal(deliveryDailyNumberLabel({ publicCode: "AST-20260517-009", storeDailyNumber: 9 }), "N. dia 009");
});

test("uses daily number as primary label when available", () => {
  assert.equal(deliveryPrimaryCodeLabel({ publicCode: "AST-20260517-009", storeDailyNumber: 9 }), "N. dia 009");
});

test("falls back to public code for older deliveries without daily number", () => {
  assert.equal(deliveryDailyNumberLabel({ publicCode: "ENT-000001" }), null);
  assert.equal(deliveryPrimaryCodeLabel({ publicCode: "ENT-000001" }), "ENT-000001");
});
